cores/esp8266/WString.cpp

```diff
@@ -677,10 +655,11 @@ String String::substring(unsigned int left, unsigned int right) const {
         return out;
     if (right > len())
         right = len();
-    char temp = buffer()[right];  // save the replaced character
-    wbuffer()[right] = '\0';
-    out = wbuffer() + left;  // pointer arithmetic
-    wbuffer()[right] = temp;  //restore character
+    char *wbuffer = this->wbuffer();
```

Comment:
Same comment as above.

Also, for these two changes have you seen any code size difference?  wbuffer() is a pretty simple call and I think it has a good chance of being inlined to a simple reference to the string array...